x-pack/test/functional/services/ml/job_annotations_table.ts

```diff
@@ -341,5 +341,39 @@ export function MachineLearningJobAnnotationsProvider({ getService }: FtrProvide
         await this.setAnnotationText(text);
       });
     }
+
+    public async assertAnnotationsDelayedDataChartActionExists(annotationId: string) {
+      await retry.tryForTime(1000, async () => {
+        await testSubjects.existOrFail('mlAnnotationsActionViewDatafeed');
+      });
+    }
+
+    public async clickAnnotationsDelayedDataChartAction(annotationId: string) {
+      await retry.tryForTime(10 * 1000, async () => {
+        const annotationsTable = await testSubjects.find('mlAnnotationsTable', 30 * 1000);
+        const collapsedActionsButton = await annotationsTable.findByCssSelector(
+          `[data-test-subj="euiCollapsedItemActionsButton"]`
+        );
+        collapsedActionsButton.click();
+      });
+    }
+
+    public async assertDelayedDataChartExists(annotationId: string) {
+      await this.clickAnnotationsDelayedDataChartAction(annotationId);
+      await this.assertAnnotationsDelayedDataChartActionExists(annotationId);
+      await testSubjects.clickWhenNotDisabled('mlAnnotationsActionViewDatafeed');
+
+      await testSubjects.existOrFail('mlAnnotationsViewDatafeedFlyout');
+
+      await testSubjects.existOrFail('mlAnnotationsViewDatafeedFlyoutTitle');
+
+      const title = await testSubjects.getVisibleText('mlAnnotationsViewDatafeedFlyoutTitle');
+      expect(title).to.contain(
+        `Datafeed chart for`,
+        `Expected annotations flyout title to be 'Datafeed chart for {jobId}' but got ${title}`
+      );
```

Comment:
1. Could wrap these lines into a separate method for future re-use, something like `openDatafeedChartFlyout`.
2. Also, I'd suggest to call this separate method from the test file, such that this method `assertDelayedDataChartExists` really only asserts the chart existence in the flyout. The test file calls would then look something like
    ```
    await ml.jobAnnotations.openDatafeedChartFlyout(annotationId);
    await ml.jobAnnotations.assertDelayedDataChartExists(annotationId);
    ```

3. The assertion message already mentions a `{jobId}`, might be good to also pass in the job id and include it in the assertion?
    ```
    expect(title).to.eql(`Datafeed chart for ${jobId}`, `Expected annotations flyout title to be 'Datafeed chart for ${jobId}' but got ${title}`
    ```

Reply:
Updated in ml-add-tests